### PR TITLE
Python processor SDK: Metrics Support

### PR DESCRIPTION
**Pull request by rjenkins**
Completes: STR-3433

This PR adds support for OTel metrics in the Python processor SDK. We're following the existing architectural patterns for logs and traces so not too much new to mention on the design approach front. There is a significant amount of room here for lazy ArcMutex construction and desctruction optimization use the `_raw` approach as attributes are pervasive in the metrics model. Metrics also introduces metadata and exemplars which may often not be mutated. Further it's likely in many cases metric data itself may not need mutation as well so I expect we'll be able to greatly improve performance in a follow up change set. While on the note of performance, I haven't spent much time yet looking at the numbers, with the idea being landing the functionality first.

Considering https://github.com/streamfold/rotel/issues/114. This PR should provide support for these use cases and many others. 

Following up on the PR we expect to land the following:
* Updates to `rotel-sdk` for .pyi type data
* Additional unit tests for edge cases and mentioned use cases in issue 114
* Performance optimizations
* Out of the box processor support.

### CODE REVIEW

File: rotel_python_processor_sdk/src/model/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+/// RHistogramDataPoint is a single data point in a timeseries that describes the
+/// time-varying values of a Histogram.
+#[derive(Debug, Clone)]
+pub struct RHistogramDataPoint {
+    pub attributes: Arc<Mutex<Vec<RKeyValue>>>,
+    pub start_time_unix_nano: u64,
+    pub time_unix_nano: u64,
+    pub count: u64,
+    pub sum: Option<f64>,
+    // TODO these should be an ARC?
```
Comment (mheffner) [inline rotel_python_processor_sdk/src/model/metrics.rs:109]: Are these important enough for having a followup issue to track?

File: rotel_python_processor_sdk/src/py/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,2170 @@
...
+            DataPointFlags::NoRecordedValueMask => Ok(1),
+        }
+    }
+}
+
+impl From<u32> for DataPointFlags {
+    fn from(value: u32) -> Self {
+        match value {
+            1 => DataPointFlags::NoRecordedValueMask,
+            _ => DataPointFlags::DoNotUse,
```
Comment (mheffner) [inline rotel_python_processor_sdk/src/py/metrics.rs:2158]: If you want these to be explicit, you could also use the TryFrom trait: https://doc.rust-lang.org/std/convert/trait.TryFrom.html

**Review by mheffner — APPROVED** [review]
Looks good, given the amount of test coverage I mostly skimmed. Happy to look at anything more in depth if it helps.

:+1: on seeing what the general perf looks like
